src/sync/chase_lev.rs

```diff
@@ -158,68 +159,91 @@ impl<T> Deque<T> {
     unsafe fn push(&self, data: T) {
         let guard = epoch::pin();
 
-        let mut b = self.bottom.load(SeqCst);
-        let t = self.top.load(SeqCst);
-        let mut a = self.array.load(SeqCst, &guard).unwrap();
+        let mut b = self.bottom.load(Relaxed);
+        let t = self.top.load(Acquire);
+        let mut a = self.array.load(Relaxed, &guard).unwrap();
+        
         let size = b - t;
         if size >= (a.size() as isize) - 1 {
```

Comment:
Not sure what to make of this, but the original paper uses
`size >= a.size() - 1`
while the new paper uses
`size > a.size() - 1`